pkg/inventory/web/client.go

```diff
@@ -228,6 +276,9 @@ func (r *WatchReader) start() {
 	go func() {
 		defer func() {
 			_ = r.webSocket.Close()
+			r.started = false
+			r.done = true
+			r.handler.End()
```

Comment:
Moved to _defer_ to ensure state is always updated and handler notified that the watch ended.